src/main/java/util/xcmailr/XcMailrApi.java

```diff
@@ -25,45 +25,103 @@
     public final static String TEMPORARY_CONFIG_FILE_PROPERTY_NAME = "xcmailr.temporaryConfigFile";
 
     /**
-     * Retrieves the instance of xcmailr configuration for the current thread.
+     * Retrieves the {@link XcMailrConfiguration} instance of the current thread.
      * 
-     * @return the configuration instance for the current thread
+     * @return the XcMailrConfiguration instance of the current thread
      */
     public static XcMailrConfiguration getConfiguration()
     {
         // the property needs to be a valid URI in order to satisfy the Owner framework
         if (null == ConfigFactory.getProperty(TEMPORARY_CONFIG_FILE_PROPERTY_NAME))
         {
-            ConfigFactory.setProperty(TEMPORARY_CONFIG_FILE_PROPERTY_NAME, "file:this/path/should/never/exist/noOneShouldCreateMe.properties");
+            ConfigFactory.setProperty(TEMPORARY_CONFIG_FILE_PROPERTY_NAME,
+                                      "file:this/path/should/never/exist/noOneShouldCreateMe.properties");
         }
         return CONFIGURATION.computeIfAbsent(Thread.currentThread(), key -> {
             return ConfigFactory.create(XcMailrConfiguration.class);
         });
     }
 
+    /**
+     * Reset configuration properties for thread. Needed for unit tests to enable reset of configurations for individual
+     * test methods, which belong to one test class
+     */
+    protected static void resetConfigurationsForThread()
```

Comment:
```suggestion
    static void resetConfigurationsForThread()
```

The default visibility modifier should be enough.